test/common/http/header_map_impl_test.cc

```diff
@@ -1189,6 +1192,25 @@ TEST(HeaderMapImplTest, TestHeaderMapImplyCopy) {
   EXPECT_EQ("bar", baz.get(LowerCaseString("foo"))->value().getStringView());
 }
 
+// Make sure 'host' -> ':authority' auto translation only occurs for request headers.
+TEST(HeaderMapImplTest, HostHeader) {
+  TestRequestHeaderMapImpl request_headers{{"host", "foo"}};
+  EXPECT_EQ(request_headers.size(), 1);
+  EXPECT_EQ(request_headers.get_(":authority"), "foo");
+
+  TestRequestTrailerMapImpl request_trailers{{"host", "foo"}};
+  EXPECT_EQ(request_trailers.size(), 1);
+  EXPECT_EQ(request_trailers.get_("host"), "foo");
```

Comment:
Can you add a test that checks if `HostLegacy` is set in this header map (and the following response headers/trailers)? I think it probably is.

Reply:
HostLegacy is just a string which == "host"? do you mean you want me to modify the test to use the constant?

Reply:
It's probably all good, just wanted it to be clear the same goes for `Host()` and `HostLegacy()` in case users use those instead of the strings. 

I only saw users translating `Host` to `HostLegacy()` header, never the reverse anyway.